test_runner/regress/test_readonly_node.py

```diff
@@ -169,23 +169,24 @@ def generate_updates_on_main(
             )
         return last_flush_lsn
 
-    def trigger_gc_and_select(env: NeonEnv, ep_static: Endpoint):
+    def trigger_gc_and_select(env: NeonEnv, ep_static: Endpoint, ctx: str):
         """
         Trigger GC manually on all pageservers. Then run an `SELECT` query.
         """
         for shard, ps in tenant_get_shards(env, env.initial_tenant):
             client = ps.http_client()
             gc_result = client.timeline_gc(shard, env.initial_timeline, 0)
+            # Note: cannot assert on `layers_removed` here because it could be layers
+            # not guarded by the lease. Rely on successful execution of the query instead.
             log.info(f"{gc_result=}")
 
-            assert (
```

Comment:
Alternatively, if there is a way to precisely control that our updates on main does not cause new layers to be removed in GC, we could add back this check.

_Note: We do need to generate updates on the main branch after the static endpoint has been created so these layers actually has a chance to be GC-ed if there are not guarded by leases_.

https://github.com/neondatabase/neon/blob/212a79005c463596bc6341afcb57b7593a203c44/test_runner/regress/test_readonly_node.py#L202-L216